arm/Microsoft.Network/firewallPolicies/ruleCollectionGroups/deploy.bicep

```diff
@@ -0,0 +1,41 @@
+@description('Required. Name of the Firewall Policy.')
+param firewallPolicyName string
+
+@description('Required. The name of the rule collection group to deploy')
+param name string
+
+@description('Required. Priority of the Firewall Policy Rule Collection Group resource.')
+param priority int
+
+@description('Optional. Group of Firewall Policy rule collections.')
+param ruleCollections array = []
+
+@description('Optional. Customer Usage Attribution ID (GUID). This GUID must be previously registered')
+param cuaId string = ''
+
+resource firewallPolicy 'Microsoft.Network/firewallPolicies@2021-03-01' existing = {
+  name: firewallPolicyName
+}
+
+module pid_cuaId '.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
```

Comment:
Lets move that up above the `existing` resource to keep that one close together with the resource that is actually referencing it

Reply:
done